lambdas/user-list-search-corpus-parser-hydration/src/parserRequest.ts

```diff
@@ -0,0 +1,191 @@
+import { ParserDocumentFields, ParserResult } from './types';
+import { config } from './config';
+import fetchRetry from 'fetch-retry';
+import * as Sentry from '@sentry/aws-serverless';
+import { stripHtml } from 'string-strip-html';
+
+const newFetch = fetchRetry(fetch);
+
+/**
+ * Make a request to the parser service, and convert the response
+ * into the fields expected by the search index mapping.
+ * @param url the URL of the content
+ * @returns fields which can be indexed in the language-specific corpus
+ * indices
+ */
+export async function parserRequest(
+  url: string,
+): Promise<ParserDocumentFields | undefined> {
+  const options = {
+    refresh: '0',
+    images: '0',
+    videos: '0',
+    createIfNone: '1',
+    enableItemUrlFallback: '1',
+    output: 'regular',
+  };
+  const queryParams = new URLSearchParams({
+    ...options,
+    url,
+  });
+
+  const response = await newFetch(
+    `${config.parserEndpoint}?${queryParams.toString()}`,
+    {
+      signal: AbortSignal.timeout(config.parser.timeout * 1000),
+      retryDelay: 2000,
+      retryOn: async function (attempt, error, response) {
+        if (attempt > 3) return false;
+        // Could still be resolving
+        if (error == null) {
+          try {
+            const res = await response.clone().json();
+            if (res.resolved_id === '0' || res.resolved_id == null) {
+              return true;
+            }
+          } catch {
+            return true;
+          }
+        }
+      },
+    },
+  );
+  if (!response.ok) {
+    Sentry.captureEvent({
+      message: 'Request to parser not ok',
+      request: {
+        url: config.parserEndpoint,
+        query_string: queryParams.toString(),
+        method: 'get',
+        data: {
+          status: response.status,
+          statusText: response.statusText,
+        },
+      },
+    });
+    return undefined;
+  }
+  const parserResult = await response.json();
+  if (!parserResult.item_id) {
+    Sentry.captureEvent({
+      message: 'Request to parser returned null item_id',
+      request: {
+        url: config.parserEndpoint,
+        query_string: queryParams.toString(),
+        method: 'get',
+        data: {
+          result: parserResult,
+        },
+      },
+    });
+    return undefined;
+  }
+  return parserResultToDoc(parserResult);
+}
+
+/**
+ * Remove HTML tags and unnecessary HTML data from the parser response.
+ * This is likely not perfectly extracting just the relevant text blobs,
+ * but it is good enough for helping search relevance.
+ * @param article article response from the parser
+ * @returns text content with HTML removed, or undefined if invalid/empty
+ * article was passed.
+ */
+export function cleanExtractedText(
+  article: string | undefined,
+): string | undefined {
+  if (article == null || article == '') {
+    return undefined;
+  }
+  // e.g. <!--IMG_1-->, <!--VIDEO_13-->
+  const mediaCommentRegex = /\<\!--(?:IMG|VIDEO)_\d+--\>/g;
```

Comment:
Cant we make a text request that does not include the comments?

Reply:
yes, but then it includes them as nodes. It's easier to strip out the comments IMO

Reply:
gotcha